interface/super/edit_globals.php

```diff
@@ -187,55 +188,73 @@ function checkBackgroundServices(){
   // Check the current status of Audit Logging
   $auditLogStatusFieldOld = $GLOBALS['enable_auditlog'];
 
+  /*
+   * Compare form values with old database values.
+   * Only save if values differ. Improves speed.
+   */
+  
+  // Get all the globals from DB
+  $old_globals = $GLOBALS['adodb']['db']->getAssoc('SELECT gl_name, gl_index, gl_value FROM `globals` ORDER BY gl_name, gl_index',false,true);
+  
   $i = 0;
   foreach ($GLOBALS_METADATA as $grpname => $grparr) {
     foreach ($grparr as $fldid => $fldarr) {
       list($fldname, $fldtype, $flddef, $flddesc) = $fldarr;
 	  if($fldtype == 'pwd'){
-	  $pass = sqlQuery("SELECT gl_value FROM globals WHERE gl_name = '$fldid'");
-	  $fldvalueold = $pass['gl_value'];
+        $pass = sqlQuery("SELECT gl_value FROM globals WHERE gl_name = '$fldid'");
```

Comment:
You're right, it is. I just felt not right touching it. We can optimize it later.
N.B. If you worry about extra SELECTs I know a place that produces 45 similar selects in one request:

`SELECT * FROM lang_definitions JOIN lang_constants ON lang_definitions.cons_id = lang_constants.cons_id WHERE lang_id='1' AND constant_name = 'Paper Size' LIMIT 1`

:)